api/v1beta1/common.go

```diff
@@ -25,44 +25,107 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-// NodeSection is a specification of the node attributes
+// AnsibleOpts defines a logical grouping of Ansible related configuration options.
+type AnsibleOpts struct {
+	// AnsibleUser SSH user for Ansible connection
+	// +kubebuilder:validation:Optional
+	AnsibleUser string `json:"ansibleUser,omitempty"`
+
+	// AnsibleHost SSH host for Ansible connection
+	// +kubebuilder:validation:Optional
+	AnsibleHost string `json:"ansibleHost,omitempty"`
+
+	// AnsiblePort SSH port for Ansible connection
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:number"}
+	AnsiblePort int `json:"ansiblePort,omitempty"`
+
+	// AnsibleVars for configuring ansible
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Schemaless
+	AnsibleVars map[string]json.RawMessage `json:"ansibleVars,omitempty"`
+
+}
+
+// NodeSection defines the top level attributes inherited by nodes in the CR.
 type NodeSection struct {
 
+	// HostName - node name
+	// +kubebuilder:validation:Optional
+	HostName string `json:"hostName,omitempty"`
+
+	// NetworkAttachments is a list of NetworkAttachment resource names to pass to the ansibleee resource
+	// which allows to connect the ansibleee runner to the given network
 	// +kubebuilder:validation:Optional
+	NetworkAttachments []string `json:"networkAttachments,omitempty"`
```

Comment:
NetworkAttachments should be dropped from NodeSection b/c it can't be overwritten on a per-node level